inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -17,55 +17,241 @@
+import static org.apache.commons.lang3.StringUtils.substring;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.stream.IntStream;
+
+import org.apache.wicket.feedback.IFeedback;
+import org.apache.wicket.markup.head.CssHeaderItem;
+import org.apache.wicket.markup.head.IHeaderResponse;
+import org.apache.wicket.markup.head.JavaScriptHeaderItem;
+import org.apache.wicket.markup.html.WebComponent;
+import org.apache.wicket.markup.html.basic.Label;
 import org.apache.wicket.markup.html.form.CheckBox;
 import org.apache.wicket.markup.html.form.Form;
 import org.apache.wicket.markup.html.form.NumberTextField;
 import org.apache.wicket.model.CompoundPropertyModel;
 import org.apache.wicket.model.IModel;
 import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wicketstuff.event.annotation.OnEvent;
 
+import de.agilecoders.wicket.webjars.request.resource.WebjarsCssResourceReference;
+import de.agilecoders.wicket.webjars.request.resource.WebjarsJavaScriptResourceReference;
 import de.tudarmstadt.ukp.clarin.webanno.api.JCasProvider;
 import de.tudarmstadt.ukp.clarin.webanno.api.annotation.action.AnnotationActionHandler;
 import de.tudarmstadt.ukp.clarin.webanno.api.annotation.model.AnnotatorState;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.rendering.event.RenderAnnotationsEvent;
 import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaAjaxButton;
 import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaModelAdapter;
 import de.tudarmstadt.ukp.clarin.webanno.ui.annotation.AnnotationPage;
 import de.tudarmstadt.ukp.clarin.webanno.ui.annotation.sidebar.AnnotationSidebar_ImplBase;
+import de.tudarmstadt.ukp.inception.log.EventRepository;
+import de.tudarmstadt.ukp.inception.log.model.LoggedEvent;
 import de.tudarmstadt.ukp.inception.recommendation.api.RecommendationService;
 import de.tudarmstadt.ukp.inception.recommendation.api.model.Preferences;
+import de.tudarmstadt.ukp.inception.recommendation.log.RecommenderEvaluationResultEventAdapter.Details;
+
+public class RecommendationSidebar extends AnnotationSidebar_ImplBase {
+	private static final long serialVersionUID = 4306746527837380863L;
+
+	private Logger log = LoggerFactory.getLogger(getClass());
+	private WebComponent chartContainer;
+
+	private @SpringBean RecommendationService recommendationService;
+	private @SpringBean EventRepository eventRepo;
+
+	IModel<AnnotatorState> aModel;
+
+	int maxPointsToPlot = 50;
+
+	public RecommendationSidebar(String aId, IModel<AnnotatorState> aModel, AnnotationActionHandler aActionHandler,
+			JCasProvider aJCasProvider, AnnotationPage aAnnotationPage) {
+		super(aId, aModel, aActionHandler, aJCasProvider, aAnnotationPage);
+		this.aModel = aModel;
```

Comment:
Eclipse allows configuring this default prefix "a" under Preferences -> Java -> Code Style set the "prefix list" column in the "parameters" row to "a".

Reply:
@Rentier  I get it why it should be this.model instead of this.aModel. But must every argument to every function start with an a or only the one where we have another object of the same type as in the argument?

Reply:
All the arguments. Then you can also leave out the `this`